presto-main/src/main/java/com/facebook/presto/execution/SetSessionTask.java

```diff
@@ -54,12 +57,18 @@ else if (propertyName.getParts().size() == 2) {
 
         Type type = propertyMetadata.getSqlType();
 
-        Object objectValue = evaluatePropertyValue(statement.getValue(), type, session, metadata);
-        String value = serializeSessionProperty(type, objectValue);
+        try {
+            Object objectValue = evaluatePropertyValue(statement.getValue(), type, session, metadata);
+            String value = serializeSessionProperty(type, objectValue);
 
-        // verify the SQL value can be decoded by the property
-        metadata.getSessionPropertyManager().decodeProperty(propertyName.toString(), value, propertyMetadata.getJavaType());
+            // verify the SQL value can be decoded by the property
+            metadata.getSessionPropertyManager().decodeProperty(propertyName.toString(), value, propertyMetadata.getJavaType());
 
-        stateMachine.addSetSessionProperties(propertyName.toString(), value);
+            stateMachine.addSetSessionProperties(propertyName.toString(), value);
+        }
+        catch (SemanticException e) {
```

Comment:
As part of fixing handling of table properties, we had to make the code that session properties used more generic. I'll update the commit message to reflect this.
Also, the error message does include details of the caught exception, so it should be pretty clear what went wrong.